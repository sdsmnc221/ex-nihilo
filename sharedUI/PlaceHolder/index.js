import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import styled, { css } from 'styled-components';

const Wrapper = styled.View`
  background-color: ${({ color }) => color};
  width: ${({ size, width }) => width || size}px;
  height: ${({ size, height }) => height || size}px;
  border-radius: ${({ round, size }) => (round ? size : 0)}px;
  justify-content: center;
  align-items: center;

  /* ${({ size }) => css`
			width: ${size}px;
			height: ${size}px;
		`} */
`;

const PlaceHolder = ({ color, size, width, height, round, text }) => (
	<Wrapper color={color} size={size} width={width} height={height} round={round}>
		{text && <Text>{text}</Text>}
	</Wrapper>
);

PlaceHolder.propTypes = {
	color: PropTypes.string,
	size: PropTypes.number,
	width: PropTypes.number,
	height: PropTypes.number,
	round: PropTypes.bool,
	text: PropTypes.string,
};

PlaceHolder.defaultProps = {
	color: '#565656',
	size: 50,
	width: undefined,
	height: undefined,
	round: false,
	text: undefined,
};

export default PlaceHolder;
