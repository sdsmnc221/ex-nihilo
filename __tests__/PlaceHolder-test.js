/**
 * ASIMPLE TEST: component PlaceHolder rendered correctly.
 */

import 'react-native';
import React from 'react';

import PlaceHolder from '../src/sharedUI/PlaceHolder';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders PlaceHolder correctly', () => {
	const tree = renderer.create(<PlaceHolder />).toJSON();
	expect(tree).toMatchSnapshot();
});
