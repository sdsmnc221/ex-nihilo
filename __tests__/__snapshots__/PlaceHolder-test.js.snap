// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlaceHolder correctly 1`] = `
<View
  color="#565656"
  round={false}
  size={50}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#565656",
        "borderRadius": 0,
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      },
    ]
  }
/>
`;
